python/lsst/obs/hsc/hscMapper.py

```diff
@@ -127,7 +139,12 @@ def map(self, datasetType, dataId, write=False):
         """
         copyId = dataId.copy()
         copyId.pop("flags", None)
-        return super(HscMapper, self).map(datasetType, copyId, write=write)
+        location = super(HscMapper, self).map(datasetType, copyId, write=write)
+
+        if 'flags' in dataId:
+            location.getAdditionalData().set('flags', dataId['flags'])
+
+        return location
```

Comment:
I agree.  History preservation question: should I change the commit message, or just add a note to it?